projects/packages/forms/src/blocks/contact-form/editor.scss

```diff
@@ -42,69 +42,111 @@
 	}
 
 	.block-editor-block-list__layout {
-		display: flex;
-		flex-wrap: wrap;
-		justify-content: flex-start;
-		flex-direction: row;
-		gap: var(--wp--style--block-gap, 1.5rem);
-
-		.wp-block {
-			flex: 0 0 100%;
-			margin-top: 0;
-			margin-bottom: 0;
+		// In a constrained layout, the blocks take up a 'content size' that's defined by the theme or user.
+		// Blocks are centered using `margin: auto;`.
+		// The percentage widths need to be multiplied by the content size and additional right margin is used to keep
+		// the blocks aligned to the left of the container.
+		// Field block currently don't support alignment, so only the default 'left' alignment is supported.
+		// TODO: consider RTL.
+		&.is-layout-constrained > * {
+			&.jetpack-field__width-25 {
+				max-width: calc(var(--wp--style--global--content-size) * 0.25);
+				margin-right: calc(50% + (var(--wp--style--global--content-size) * 0.25)) !important;
```

Comment:
This won't work if a custom content width is set on the Group block. When that happens, max width is set to a specific value, not a variable. I can't think of a reasonable way to fix this that doesn't involve some JS trickery 😅 

The other thing is, the margin-right won't play well with different content justifications:

(this example shows a Group with 300px justified right content where both Name and Email fields are set to 75% width; Message has no width set)

<img width="896" alt="Screenshot 2025-02-17 at 3 21 24 pm" src="https://github.com/user-attachments/assets/813291c2-a16f-4e53-91b4-f891641f77a4" />

Reply:
Yeah, I've only had a quick look into this, and I had a feeling it wouldn't work all that well.

I know some blocks like image manage to make it work, but do so by keeping the `<figure>` wrapper at full width. It might be possible to try something like that, but I'm also keen not to introduce something that's hard to support or might end up buggy.

Reply:
I've removed width support in constrained layouts for now 👍